plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -387,7 +387,7 @@ def sweep(self, arguments: List[str]) -> None:
 
                 except Exception as e:
                     state = optuna.trial.TrialState.FAIL
-                    study.tell(trial=trial, state=state, values=values)
+                    study.tell(trial=trial, state=state, values=None)
```

Comment:
could we add a check here and only set `values=None` if "nan" exists in values? `study.tell` can fail in many other ways, and could potentially also caused by other bugs so i think it'd better to be more specific here.

Reply:
Sure, I can add this

Reply:
Maybe a more specific way to handle this problem prior to `optuna 3.0` is to check `NaN` right after we cast every return values to float. If `NaN` is found, we can tell `optuna` this study is `FAIL` directly. However, I think even in that case, this change is still valid.